src/lib/Simplify.hs

```diff
@@ -764,12 +764,10 @@ projectDictMethod d i = do
       case i of
         0 -> return method
         _ -> error "ExplicitDict only supports single-method classes"
-    DictCon (ExplicitMethods methodImplName params) -> do
-      SpecializedDictBinding (SpecializedDictDef _ fs) <- lookupEnv methodImplName
-      let f = fs !! i
-      TopFunBound _ (SpecializedTopFun (IxMethodSpecialization _ (Abs bs dict'))) <- lookupAtomName f
-      dict'' <- applySubst (bs @@> map SubstVal params) dict'
-      projectDictMethod dict'' i
+    DictCon (ExplicitMethods sd params) -> do
+      SpecializedDictBinding (SpecializedDict (Abs bs dict) _) <- lookupEnv sd
+      dict' <- applySubst (bs @@> map SubstVal params) dict
+      projectDictMethod dict' i
```

Comment:
I am a bit surprised here. Isn't the second component of `SpecializedDict` supposed to hold the names of specialized methods? Why can't we just look them up and use them here?

Reply:
Not anymore. The second component is `Maybe [NaryLamExpr n]`. During simplification, we can't be sure that the field will be non-Nothing. Probably deserves a comment.